src/util.cc

```diff
@@ -134,4 +134,20 @@ namespace roboptim
       }
     return true;
   }
+
+  inline bool is_malloc_allowed_update(bool update, bool new_value = false)
```

Comment:
`bool update = false`? Thus default would simply return the current value. Or should we keep `update` explicitly set?